frontend/src/contexts/network-context.tsx

```diff
@@ -25,10 +25,11 @@ export const networkContext = React.createContext<NetworkContext>(null!);
 const newNode = {
   id: "0",
   data: {
-  name: "new node",
-  label:"new node",
-  size: 10,
+    id: "0",
```

Comment:
i see duplicates keys (id).

Reply:
The `data` property, has a field `id` which is the `id` here. I made the `id` field mandatory (it was an optional type previously). The formatting makes it look like a duplicate, its a field under data.

Discussions on wether the field is redundant is a different matter. (I'm indifferent)
I think if we want separation of concern between node properties for the canvas (id, type, position) and node data itself, we can leave the second `id` field to reference the node.

There's a specific instance where separation of concern can simplify the editing fnc but its not a priority at the moment so we can leave the `id` as is for now.